commit 601ba1fca3ab86e5d3dc92817a98fdfaab440e9c
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:22:06 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-allow-clazy-and-gcclibcxx.diff

diff --git a/src/port1.0/portconfigure.tcl b/src/port1.0/portconfigure.tcl
index 2ac80129c..680304412 100644
--- a/src/port1.0/portconfigure.tcl
+++ b/src/port1.0/portconfigure.tcl
@@ -81,13 +81,13 @@ proc portconfigure::should_add_stdlib {} {
     if {${configure.cxx_stdlib} eq ""} {
         return 0
     }
-    if {[string match *clang* ${configure.cxx}]} {
+    if {[string match *clang* ${configure.cxx}] || [string match *clazy* [option configure.cxx]]} {
         return 1
     }
     # GCC also supports -stdlib starting with GCC 10 (and devel), but
     # not with PPC builds
     global configure.build_arch
-    if {[string match *g*-mp-* ${configure.cxx}]
+    if {[string match */g*-mp-* ${configure.cxx}]
             && ${configure.build_arch} ni {ppc ppc64}} {
         # Do not pass stdlib to gcc if it is MacPorts custom macports-libstdc++ setting
         # as gcc does not uderstand this. Instead do nothing, which means gcc will
@@ -375,6 +375,7 @@ proc portconfigure::configure_start {args} {
         {^llvm-gcc-4\.2$}                          {Xcode LLVM-GCC 4.2}
         {^macports-clang$}                         {MacPorts Clang (port select)}
         {^macports-clang-(\d+(?:\.\d+)?)$}         {MacPorts Clang %s}
+        {^macports-clazy-(\d+(?:\.\d+)?)$}         {MacPorts Clazy %s}
         {^macports-gcc$}                           {MacPorts GCC (port select)}
         {^macports-gcc-(\d+(?:\.\d+)?)$}           {MacPorts GCC %s}
         {^macports-llvm-gcc-4\.2$}                 {MacPorts LLVM-GCC 4.2}
@@ -747,6 +748,7 @@ proc portconfigure::compiler_port_name {compiler} {
     set valid_compiler_ports {
         {^apple-gcc-(\d+)\.(\d+)$}                                                    {apple-gcc%s%s}
         {^macports-clang-(\d+(?:\.\d+)?)$}                                            {clang-%s}
+        {^macports-clazy-(\d+(?:\.\d+)?)$}                                            {clazy-%s}
         {^macports-(llvm-)?gcc-(\d+)(?:\.(\d+))?$}                                    {%sgcc%s%s}
         {^macports-(mpich|openmpi|mpich-devel|openmpi-devel)-default$}                {%s-default}
         {^macports-(mpich|openmpi|mpich-devel|openmpi-devel)-clang$}                  {%s-clang}
@@ -1558,6 +1560,15 @@ proc portconfigure::configure_get_compiler {type {compiler {}}} {
             objcxx  { return ${prefix_frozen}/bin/clang++${suffix} }
             cpp     { return ${prefix_frozen}/bin/clang-cpp${suffix} }
         }
+    } elseif {[regexp {^macports-clazy(-\d+\.\d+)?$} $compiler -> suffix]} {
+        set suffix "-mp${suffix}"
+        switch $type {
+            cc      -
+            objc    { return ${prefix_frozen}/bin/clang${suffix} }
+            cxx     -
+            objcxx  { return ${prefix_frozen}/bin/clang++${suffix} }
+            cpp     { return ${prefix_frozen}/bin/clang-cpp${suffix} }
+        }
     } elseif {[regexp {^macports-gcc(-\d+(?:\.\d+)?)?$} $compiler -> suffix]} {
         if {$suffix ne ""} {
             set suffix "-mp${suffix}"
